coreax/solvers/recombination.py

```diff
@@ -0,0 +1,570 @@
+    \end{gather}
+
+.. note::
+    The source dataset is, by definition, a solution to the linear-program that is not
+    necessarily a BFS. Hence, one may consider the fundamental problem of recombination
+    as that of finding a BFS given a solution that is not a BFS.
+
+Basic feasible solutions to the linear-program above are of the form
+:math:`\mathbf{\hat{w}} = \{\hat{w}_1, \dots, \hat{w}_{m^\prime}, \mathbf{0}\}`; I.E.
+BFSs are feasible solutions with :math:`n-m^\prime` weights equal to zero. Given a BFS,
+the reduced measure (the coresubset) can be constructed by explicitly removing the nodes
+associated with each zero valued (implicitly removed) weight
+
+.. math::
+    \begin{gather}
+    \hat{\nu}_{m^\prime} = \sum_{i \in I} \hat{w_i} \delta_{x_i},\\
+    I = \{i \mid \hat{w_i} \neq 0\, \forall i \in \{1, \dots, n\}\}.
+    \end{gather}
+
+Due to Tchakaloff's theorem, which follows from Caratheodory's convex hull theorem, we
+know there always exists a basic-feasible solution to the linear-program, with at most
+:math:`m^\prime = \text{dim}(\text{span}(\Phi))` non-zero weights. Hence, we have an
+upper bound on the size of a coresubset, controlled by the choice of test-functions.
+
+.. note::
+    A basic feasible solution (coresubset produced by recombination) is non-unique. In
+    fact, there exists :math:`\binom{n}{m^\prime}` basic feasible solutions
+    (coresubsets) for the described linear-program. In the context of Coreax, this means
+    that a :class:`RecombinationSolver` is unlikely to ever be truly invariant to the
+    presence of padding (see :class:`~coreax.solvers.PaddingInvariantSolver`). I.E. the
+    padded problem may have an equivalent, but different BFS than the unpadded problem.
+
+Canonically, recombination is used for reducing the support of a quadrature/cubature
+measure, against which integration of any function :math:`f \in \text{span}(\Phi)`
+is identical to integration against a "target" (potentially continuous) measure
+:math:`\mu`.
+"""
+
+import math
+from collections.abc import Callable
+from typing import Generic, Literal, NamedTuple, Optional, TypeVar, Union
+
+import jax
+import jax.numpy as jnp
+import jax.scipy as jsp
+import jax.tree_util as jtu
+from jaxtyping import Array, DTypeLike, Real, Shaped
+from typing_extensions import override
+
+from coreax import Coresubset, Data
+from coreax.solvers.base import CoresubsetSolver
+
+_Data = TypeVar("_Data", bound=Data)
+_State = TypeVar("_State")
+Omega = TypeVar("Omega")
+
+
+class RecombinationSolver(CoresubsetSolver[_Data, _State], Generic[_Data, _State]):
+    r"""
+    Solver which returns a :class:`coreax.coreset.Coresubset` via recombination.
+
+    Given :math:`m-1` explicitly provided test-functions :math:`\Phi^\prime`, a
+    recombination solver finds a coresubset with :math:`m^\prime \le m` points, whose
+    push-forward :math:`\hat\{mu}_{\m^\prime}` has the same "centre-of-mass" as the
```

Comment:
should be `:math:`\hat{\mu}_{m^\prime}`` ?

Reply:
Good spot!